src/classes/Pricelist.ts

```diff
@@ -198,112 +204,43 @@ export default class Pricelist extends EventEmitter {
     }
 
     hasPrice(sku: string, onlyEnabled = false): boolean {
-        const index = this.getIndex(sku);
-        if (index === -1) {
+        if (!this.prices[sku]) {
             return false;
         }
 
-        const match = this.prices[index];
-        return !onlyEnabled || match.enabled;
+        return this.prices[sku].enabled || !onlyEnabled;
     }
 
     getPrice(sku: string, onlyEnabled = false, generics = false): Entry | null {
-        const pSku = SKU.fromString(sku);
-        // Index of of item in pricelist
-        const index = this.getIndex(null, pSku);
-        const gindex = index === -1 && generics ? this.getIndexWithGenerics(null, pSku) : -1;
-
-        if (index === -1 && (!generics || (generics && gindex === -1))) {
-            // Did not find a match
-            return null;
-        }
-
-        let match = index !== -1 ? this.prices[index] : this.prices[gindex];
-        if (gindex !== -1) {
-            // we found a generic match for a specific sku so we are going to clone the generic entry
-            // otherwise we would mutate the existing generic entry
-            match = match.clone();
-
-            const effectMatch = this.bot.effects.find(e => pSku.effect === e.id);
-            match.name = match.name.replace('Unusual', effectMatch.name);
-            match.sku = sku;
-
-            // change any other options if needed here (possible spot for config)
+        if (this.hasPrice(sku, onlyEnabled)) {
+            return this.prices[sku];
         }
 
-        if (onlyEnabled && !match.enabled) {
-            // Item is not enabled
-            return null;
+        if (generics) {
+            const gSku = generics ? sku.replace(/;u\d+/, '') : null;
+            if (this.hasPrice(gSku, onlyEnabled)) {
+                return this.prices[gSku];
+            }
```

Comment:
I think this will not work.

Reply:
What makes you think so ?

Reply:
Sorry for the delay. I will try to come up with my guess later. Thanks btw.